netty/src/test/java/io/grpc/transport/netty/NettyClientHandlerTest.java

```diff
@@ -176,14 +198,16 @@ public void cancelShouldSucceed() throws Exception {
   }
 
   @Test
-  public void cancelBeforeStreamAssignedShouldSucceed() throws Exception {
+  public void cancelWhileBufferedShouldSucceed() throws Exception {
     handler.connection().local().maxActiveStreams(0);
     handler.write(ctx, new CreateStreamCommand(grpcHeaders, stream), promise);
     mockContext();
-    verify(stream, never()).id(any(Integer.class));
-    when(stream.id()).thenReturn(null);
-
-    handler.write(ctx, new CancelStreamCommand(stream), promise);
+    ArgumentCaptor<Integer> idCaptor = ArgumentCaptor.forClass(Integer.class);
+    verify(stream).id(idCaptor.capture());
+    when(stream.id()).thenReturn(idCaptor.getValue());
+    ChannelPromise cancelPromise = mock(ChannelPromise.class);
+    handler.write(ctx, new CancelStreamCommand(stream), cancelPromise);
+    verify(cancelPromise).setSuccess();
```

Comment:
@nmittler (I assume you meant to comment one line down) yeah it's a bit counter intuitive at first, but when you think about it, if one sends a RST_STREAM to a buffered stream one doesn't really care about that stream anymore and so failing all the promises and invoking error handling doesn't really make sense either. So we decided to just act as if the frames have been send successfully instead. Also see https://github.com/grpc/grpc-java/pull/225/files#diff-f698e86adb3b3d41fe4822f612850d80R142